test/org/sagebionetworks/bridge/validators/ConsentSignatureValidatorTest.java

```diff
@@ -0,0 +1,240 @@
+        try {
+            Validate.entityThrowingException(validator, sig);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertTrue(e.getMessage().contains(
+                    "ConsentSignature must specify imageData and imageMimeType if you specify either of them"));
+        }
+    }
+
+    @Test
+    public void jsonImageMimeTypeWithoutData() throws Exception {
+        String jsonStr = "{\n" +
+                "   \"name\":\"test name\",\n" +
+                "   \"birthdate\":\"1970-01-01\",\n" +
+                "   \"imageMimeType\":\"image/fake\"\n" +
+                "}";
+        ConsentSignature sig = BridgeObjectMapper.get().readValue(jsonStr, ConsentSignature.class);
+        try {
+            Validate.entityThrowingException(validator, sig);
+            fail("Should have thrown exception");
+        } catch(InvalidEntityException e) {
+            assertTrue(e.getMessage().contains(
+                    "ConsentSignature must specify imageData and imageMimeType if you specify either of them"));
+        }
+    }
+
+    @Test
+    public void minAgeLimitButNoBirthdate() {
+        validator = new ConsentSignatureValidator(18);
+        ConsentSignature sig = new ConsentSignature.Builder().withName("test name").build();
+        assertValidatorMessage(validator, sig, "birthdate", "cannot be missing, null, or blank");
+    }
+    
+    @Test
+    public void minAgeLimitButBirthdateTooRecent() {
+        String birthdate = DateTime.now().minusYears(18).plusDays(1).toLocalDate().toString();
+        validator = new ConsentSignatureValidator(18);
+        ConsentSignature sig = new ConsentSignature.Builder().withName("test name").withBirthdate(birthdate).build();
+        assertValidatorMessage(validator, sig, "birthdate", "too recent (the study requires participants to be 18 years of age or older).");
+    }
+    
+    @Test
+    public void minAgeLimitBirthdateOK() {
+        String birthdate = DateTime.now().minusYears(18).toLocalDate().toString();
+        validator = new ConsentSignatureValidator(18);
+        ConsentSignature sig = new ConsentSignature.Builder().withName("test name").withBirthdate(birthdate).build();
+        Validate.entityThrowingException(validator, sig);
+    }
+    
+    @Test
+    public void minAgeLimitBirthdateGarbled() {
+        validator = new ConsentSignatureValidator(18);
+        ConsentSignature sig = new ConsentSignature.Builder().withName("test name").withBirthdate("15 May 2018").build();
+        assertValidatorMessage(validator, sig, "birthdate", "is invalid (required format: YYYY-MM-DD)");
+    }
+    
+    @Test
+    public void optionalBirthdateGarbled() {
+        ConsentSignature sig = new ConsentSignature.Builder().withName("test name").withBirthdate("15 May 2018").build();
+        assertValidatorMessage(validator, sig, "birthdate", "is invalid (required format: YYYY-MM-DD)");
+    }
```

Comment:
Should we add a test case for optional birthday not specified is valid?

Reply:
Added